# Simple Regression

### Working with dataset faithful - observations of the Old Faithful geyser in the USA Yellowstone National Park
- Eruption time (in minutes) beased on the waiting period between eruptions (in minutes)
- Apply the simple linear regression model for the data set faithful, and estimate the next eruption duration if the waiting time since the last eruption has been 80 minutes.

In [1]:
head(faithful)

eruptions waiting
1 3.600     79     
2 1.800     54     
3 3.333     74     
4 2.283     62     
5 4.533     85     
6 2.883     55

In [2]:
nrow(faithful)

[1] 272

#### Linear Regression model for eruption:

In [3]:
eruption.lm <- lm(eruptions~waiting, data=faithful)
eruption.lm


Call:
lm(formula = eruptions ~ waiting, data = faithful)

Coefficients:
(Intercept)      waiting  
   -1.87402      0.07563  


#### Extract the parameters of the estimated regression equation with the coefficients function:

In [4]:
coeffs = coefficients(eruption.lm)
coeffs

(Intercept)     waiting 
-1.87401599  0.07562795

####  Fit the eruption duration using the estimated regression equation:
- Estimate the next eruption duration if the waiting time since the last eruption has been 80 minutes
- y = a + bx

In [5]:
waiting = 80
duration = coeffs[1] + coeffs[2]*waiting 
duration 

(Intercept) 
    4.17622

- If the waiting time is 80 minutes, then the eruption should last for 4.17 minutes

### Alternative solution using predict() function:

#### Wrap the waiting parameter value inside a new data frame named newWait

In [6]:
newWait = data.frame(waiting=80)
newWait

waiting
1 80

#### Apply the predict function to eruption.lm along with newdata

In [7]:
eruption.lm <- lm(eruptions~waiting, data=faithful)
predict(eruption.lm, newWait)

1 
4.17622

- Same Answer as above

In [8]:
#help(predict.lm)

### Confidence Interval for Linear Regression
- In the data set faithful, develop a 95% confidence interval of the mean eruption duration for the waiting time of 80 minutes

In [9]:
eruption.lm <- lm(eruptions~waiting, data=faithful)

In [10]:
newWait = data.frame(waiting=80)

In [11]:
predict(eruption.lm, newWait, interval="confidence")

fit     lwr      upr     
1 4.17622 4.104848 4.247592

- The 95% confidence interval of the mean eruption duration for the waiting time of 80 minutes is between 4.1048 and 4.2476 minutes.

### Prediction Interval for Linear Regression
- In the data set faithful, develop a 95% prediction interval of the eruption duration for the waiting time of 80 minutes

In [12]:
eruption.lm <- lm(eruptions~waiting, data=faithful)

In [13]:
newWait = data.frame(waiting=80)

In [14]:
predict(eruption.lm, newWait, interval="predict")

fit     lwr      upr     
1 4.17622 3.196089 5.156351

- The 95% prediction interval of the eruption duration for the waiting time of 80 minutes is between 3.1961 and 5.1564 minutes